tracee-ebpf/tracee/events.go

```diff
@@ -0,0 +1,184 @@
+package tracee
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"strconv"
+	"unsafe"
+
+	"github.com/aquasecurity/tracee/tracee-ebpf/external"
+)
+
+type RawEvent struct {
+	Ctx      context
+	Args     map[string]interface{}
+	ArgMetas []external.ArgMeta
+}
+
+// context struct contains common metadata that is collected for all types of events
+// it is used to unmarshal binary data and therefore should match (bit by bit) to the `context_t` struct in the ebpf code.
+// NOTE: Integers want to be aligned in memory, so if changing the format of this struct
+// keep the 1-byte 'Argnum' as the final parameter before the padding (if padding is needed).
+type context struct {
+	Ts       uint64
+	Pid      uint32
+	Tid      uint32
+	Ppid     uint32
+	HostPid  uint32
+	HostTid  uint32
+	HostPpid uint32
+	Uid      uint32
+	MntID    uint32
+	PidID    uint32
+	Comm     [16]byte
+	UtsName  [16]byte
+	ContID   [16]byte
+	EventID  int32
+	Retval   int64
+	StackID  uint32
+	Argnum   uint8
+	_        [3]byte //padding
+}
+
+func (t *Tracee) processEvents(done <-chan struct{}) error {
+	for dataRaw := range t.eventsChannel {
+		dataBuff := bytes.NewBuffer(dataRaw)
+		var ctx context
+		err := binary.Read(dataBuff, binary.LittleEndian, &ctx)
+		if err != nil {
+			t.handleError(err)
+			continue
+		}
+
+		rawEvent := RawEvent{
```

Comment:
Should we create a type for this ? It feels like we're adding a layer that is not necessarily needed as this type is only being used by shouldProcessEvent().

Reply:
See my previous comment

Reply:
I misunderstood you at first. You are right, we don't need it - removed.